### PR TITLE
format-udf 1.4.0 (new formula)

### PR DESCRIPTION
**Pull request by iandrewt**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

File: Formula/format-udf.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  homepage "https://github.com/JElchison/format-udf"
+  url "https://github.com/JElchison/format-udf/archive/1.4.0.tar.gz"
+  sha256 "41f259471fce19b8b716ff73498cd9e6864ffebc3855ffb5be5d763f5397d765"
+
+  def install
+    bin.install "format-udf.sh"
+  end
+
+  test do
+    system "#{bin}/format-udf.sh", "-h"
```
Comment (MikeMcQuaid) [inline Formula/format-udf.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Formula/format-udf.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class FormatUdf < Formula
+  desc "Bash script to format a block device to UDF"
+  homepage "https://github.com/JElchison/format-udf"
+  url "https://github.com/JElchison/format-udf/archive/1.4.0.tar.gz"
+  sha256 "41f259471fce19b8b716ff73498cd9e6864ffebc3855ffb5be5d763f5397d765"
+
+  def install
+    bin.install "format-udf.sh"
```
Comment (MikeMcQuaid) [inline Formula/format-udf.rb]: Might be nicer to `bin.install "format-udf.sh" => "format-udf"`

File: Formula/format-udf.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class FormatUdf < Formula
+  desc "Bash script to format a block device to UDF"
+  homepage "https://github.com/JElchison/format-udf"
+  url "https://github.com/JElchison/format-udf/archive/1.4.0.tar.gz"
+  sha256 "41f259471fce19b8b716ff73498cd9e6864ffebc3855ffb5be5d763f5397d765"
+
```
Comment (MikeMcQuaid) [inline Formula/format-udf.rb]: Can add a `bottle :unneeded` here too.

**Comment by iandrewt** [comment]
There's no way to make a test that isn't `-h` because `format-udf` is not compatible with fake devices (I've tried making a dmg with `hdiutil` but `format-udf` wouldn't touch it)

**Comment by MikeMcQuaid** [comment]
@iandrewt Ok, no worries and thanks for trying!

File: Formula/format-udf.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+class FormatUdf < Formula
+  desc "Bash script to format a block device to UDF"
+  homepage "https://github.com/JElchison/format-udf"
+  url "https://github.com/JElchison/format-udf/archive/1.4.0.tar.gz"
+  sha256 "41f259471fce19b8b716ff73498cd9e6864ffebc3855ffb5be5d763f5397d765"
+  bottle :unneeded
```
Comment (MikeMcQuaid) [inline Formula/format-udf.rb]: Sorry for pedantry: can you add a newline above this?

File: Formula/format-udf.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+  def install
+    bin.install "format-udf.sh" => "format-udf"
+  end
+
+  test do
+    system "#{bin}/format-udf.sh", "-h"
```
Comment (MikeMcQuaid) [inline Formula/format-udf.rb]: This is failing: http://bot.brew.sh/job/Homebrew%20Core%20Pull%20Requests/2600/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/test_format_udf/

Might instead want to use `assert_match` and `shell_output` with the second argument as the expected exit code.

Comment (MikeMcQuaid) [inline Formula/format-udf.rb]: Oh, also, this should be `format-udf` rather than `format-udf.sh` now.

**Comment by iandrewt** [comment]
Fixed. The test failed because I forgot to change the test block to reflect the name change

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request merged**